examples/energy-management-app/energy-management-common/water-heater/include/WhmManufacturer.h

```diff
@@ -36,7 +40,15 @@ namespace WaterHeaterManagement {
 class WhmManufacturer : public DeviceEnergyManagement::DEMManufacturerDelegate
 {
 public:
-    WhmManufacturer(WaterHeaterManagementInstance * whmInstance) { mWhmInstance = whmInstance; }
+    WhmManufacturer(WaterHeaterManagementInstance * whmInstance,
+                    ElectricalPowerMeasurement::ElectricalPowerMeasurementInstance * aEPMInstance,
+                    PowerTopology::PowerTopologyInstance * aPTInstance, DeviceEnergyManagementManager * aDEMInstance)
```

Comment:
Could the parameters be passed in by-reference instead of pointers? This would avoid having to check thereafter that the pointers are not null.